code/game/machinery/fax_machine.dm

```diff
@@ -0,0 +1,781 @@
+	var/receiving_enabled = TRUE
+	/// Whether this fax machine is locked.
+	var/locked = TRUE
+	/// Whether this fax machine can receive paperwork to process on SSeconomy ticks.
+	var/can_receive_paperwork = TRUE
+	/// Whether we have an unread message
+	var/unread_message = FALSE
+	/// The area string this fax machine is set to.
+	var/room_tag
+	/// The paper stored that we can send to admins. Reference to something in our contents.
+	var/obj/item/paper/stored_paper
+	/// The paper received that was sent FROM admins. Reference to something in our contents.
+	var/obj/item/paper/received_paper
+	/// List of all paperwork we have in this fax machine. List of references to things in our contents.
+	var/list/obj/item/paper/processed/received_paperwork
+	/// Max amount of paperwork we can hold. Any more and the UI gets less readable.
+	var/max_paperwork = 8
+	/// Cooldown between sending faxes
+	COOLDOWN_DECLARE(fax_cooldown)
+
+/obj/machinery/fax_machine/Initialize()
+	. = ..()
+	GLOB.fax_machines += src
+	set_room_tag(TRUE)
+	wires = new /datum/wires/fax(src)
+
+/obj/machinery/fax_machine/Destroy()
+	QDEL_NULL(stored_paper)
+	QDEL_NULL(received_paper)
+	QDEL_LAZYLIST(received_paperwork)
+
+	GLOB.fax_machines -= src
+	return ..()
+
+/obj/machinery/fax_machine/on_deconstruction()
+	eject_stored_paper()
+	eject_all_paperwork()
+	eject_received_paper()
+
+	return ..()
+
+/obj/machinery/fax_machine/recieving_disabled
+	can_receive_paperwork = FALSE
+
+/obj/machinery/fax_machine/full/Initialize()
+	. = ..()
+	for(var/i in 1 to max_paperwork)
+		if(LAZYLEN(received_paperwork) >= max_paperwork)
+			continue
+		LAZYADD(received_paperwork, generate_paperwork(src))
+
+/obj/machinery/fax_machine/ui_interact(mob/user, datum/tgui/ui)
+	. = ..()
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "_FaxMachine", name)
+		ui.open()
+
+/obj/machinery/fax_machine/ui_state(mob/user)
+	if(!anchored)
+		return UI_DISABLED
+	return GLOB.physical_state
+
+/obj/machinery/fax_machine/ui_data(mob/user)
+	var/list/data = list()
+
+	var/emagged = obj_flags & EMAGGED
+	var/list/all_received_paperwork = list()
+	for(var/I in 1 to length(received_paperwork))
+		var/obj/item/paper/processed/paper = received_paperwork[I]
+		var/list/found_paper_data = list()
+		found_paper_data["title"] = paper.name
+		found_paper_data["contents"] = text_preview(remove_all_tags(paper.info), MAX_DISPLAYED_PAPER_CHARS)
+		found_paper_data["required_answer"] = paper.required_question
+		found_paper_data["ref"] = REF(paper)
+		found_paper_data["num"] = I
+		all_received_paperwork += list(found_paper_data)
+	if(all_received_paperwork.len)
+		data["received_paperwork"] = all_received_paperwork
+
+	if(stored_paper)
+		var/list/stored_paper_data = list()
+		stored_paper_data["contents"] = text_preview(remove_all_tags(stored_paper.info), MAX_DISPLAYED_PAPER_CHARS)
+		stored_paper_data["ref"] = REF(stored_paper_data)
+		data["stored_paper"] = stored_paper_data
```

Comment:
There's a similar issue with `data["stored_paper"]` too. If you put a paper into the fax machine and then remove it, the 'Send A Fax' screen never clears.
Same solution as the last one though, `data["stored_paper"] = stored_paper_data` just needs to be moved out of the check:
```suggestion
	var/list/stored_paper_data = list()
	if(stored_paper)
		stored_paper_data["contents"] = text_preview(remove_all_tags(stored_paper.info), MAX_DISPLAYED_PAPER_CHARS)
		stored_paper_data["ref"] = REF(stored_paper_data)
	data["stored_paper"] = stored_paper_data
```